### ISSUE TITLE
ActionMenu keyboard navigation: pressing space key on focused menu item should trigger the action

### ISSUE THREAD

**Issue by cbliard** [primer/view_components#2253]
When a user opens the ActionMenu dropdown by pressing space or enter, the first item is focused and the focus can be moved with the arrow keys, but pressing space does not properly trigger the associated action.

This can be experienced on [LookBook ActionMenu with_actions example](https://primer.style/view-components/lookbook/inspect/primer/alpha/action_menu/with_actions/). On this one, Enter works as expected but not space key:

- Click twice on the "Trigger" button to show and hide it so that it gets the focus
- Press Down arrow key, the menu listing actions is displayed and the first menu item, "Alert", is focused (this one is a `<button>`)
- Press Space, the menu is hidden (it should have triggered the alert)
- Press Space again, the menu is shown again and the alert action is triggered (the action should not trigger at this point)

It can also be experienced on other examples like [links](https://primer.style/view-components/lookbook/inspect/primer/alpha/action_menu/links) or [opens_dialog](https://primer.style/view-components/lookbook/inspect/primer/alpha/action_menu/opens_dialog). For these two, space key never triggers the action, even on second press.

**Comment by lesliecdubs** [comment]
Thanks for the report, @cbliard!

@TylerJDev, I can confirm the behavior described in this issue. After opening `ActionMenu`, pressing "enter" will trigger the action menu item behavior, while pressing "space" does not. Can you confirm whether this is an accessibility violation (e.g., that "space" also needs to trigger the action), or if these behaviors are only required to occur on "enter"?

**Comment by TylerJDev** [comment]
Hey @cbliard, thanks for bringing this up! 

We have a [PR which should partially fix this issue](https://github.com/primer/view_components/pull/2199). This PR will allow menu items which were originally links, to be triggered with "Space" in addition to the "Enter" key. 

Here's an example of the `ActionMenu` with links, but stemming off the draft PR's branch:
* [Links example](http://primer-view-components-preview-2199.eastus.azurecontainer.io/view-components/lookbook/inspect/primer/alpha/action_menu/links)

As for the other examples, I see we don't respect the "Space" key when triggering a menu item that was originally a button. If I had to take a guess, I believe this is because we rely on the native functionality of the button to trigger the event. With buttons, space will only trigger the event on key up which only fires after the menu has already closed.

@lesliecdubs, I can confirm this is an issue, but shouldn't require much lift to fix.

**Comment by lesliecdubs** [comment]
Thanks @TylerJDev 🙇🏻‍♀️ 

I will bring this to the PVC working group to discuss and see if we can revive https://github.com/primer/view_components/pull/2199 for merge, as well as addressing the remaining work here, which sounds like it is to respect the "Space" key when triggering a menu item that was originally a button (though please correct me if I got that wrong!).

**Comment by cbliard** [comment]
Thanks a lot for fixing this!
